provisioning/WFGP-292_doc_galleon-pack.adoc

```diff
@@ -0,0 +1,164 @@
+* We could build a searchable index of the documentation (to improve its accessibility).
+ There is also work on a https://github.com/model-graph-tools[Graph representation of the WildFly management model] that could enhance the documentation experience
+
+== Implementation Plan
+
+=== Galleon
+
+* We might need to expose additional metadata for feature pack through the Galleon API (to be investigated).
+
+=== Wildscribe
+
+* Archive the project.
+
+=== Galleon Maven Plug-in
+
+When the `build-feature-pack` goal of the `org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin` is invoked (ie when the feature pack is built):
+
+. generate the model reference of the feature pack by executing `:read-resource-description` operation on the server provisioned by the plugin (that only contains the resources provided by the feature pack)
+. Store that model in a JSON file, `model.json`
+. Generate and store feature pack metadata in a `metadata.json`
+. Generate human-readable documentation based on the feature pack model.json and metadata.json files
+* Model reference is generated by delegating it to a dedicated `doc-gen` Maven project (that does not depend on Maven plugin API)
+* Log messages are extracted from the provisioned server's JBoss modules
+. Create a `-doc.zip` archive that contains:
+* `META-INF/model.json` - the Model reference (stored in a DMR format in a JSON file)
+* `META-INF/metadata.json` - the feature pack metadata
+* `doc/` - Generated HTML documentation about the feature pack. In particular:
+** `doc/index.html` - Home page of the feature pack documentation
+** `doc/reference/index.html` - Home page of the model reference of the feature pack
+. Attach this `-doc.zip` archive to the Maven project.
+
+== Security Considerations
+
+This requirement provides documentation only.
+
+[[test_plan]]
+== Test Plan
+
+Tests will be added to the Galleon Maven plugin to verify that they are able to
+properly generate this documentation archive when the feature pack is built.
+
+== Community Documentation
+
+This feature will be documented in the Galleon Maven Plugin project and is meant to be read by feature pack developers.
+
+However once, this feature is delivered, it will have a high impact on community documentation as any of the feature pack provided by the WildFly community will be able to display this documentation (either in a standalone website, ususally on github.io, or aggregated in docs.wildfly.org).
+
+In particular, once this feature is delivered, it is recommended that feature pack developers review and check their documentation (model reference, metadata, etc.) to ensure that the information are correct, accurate and helpful.
+
+== Release Note Content
+
+Feature packs that are used to provision WildFly provide now generated documentation that describes the feature pack, its layers and the model references of the subsystems provided by the feature pack.
```

Comment:
s/provide now/now provide/g

Perhaps

"describes the feature pack and its layers, along with the management API reference for the management resources provided by the feature pack."

^^^ is partly about readability, partly about my point from the overview about the unclear term "model reference" and partly about the fact that feature packs provide server kernel resources, not just subsystems.